src/transactions/CreateClaimableBalanceOpFrame.cpp

```diff
@@ -179,13 +198,19 @@ CreateClaimableBalanceOpFrame::doApply(AbstractLedgerTxn& ltx)
             innerResult().code(CREATE_CLAIMABLE_BALANCE_UNDERFUNDED);
             return false;
         }
-    }
 
-    // Create claimable balance entry
-    LedgerEntry newClaimableBalance;
-    newClaimableBalance.data.type(CLAIMABLE_BALANCE);
+        // if the issuer is creating a claimable balance, then we need to check
+        // the account flags to determine if clawback should be enabled on the
+        // balance
+        if (header.current().ledgerVersion >= 16 &&
+            (trustline.isClawbackEnabled() ||
+             (getSourceID() == getIssuer(asset) &&
```

Comment:
This logic/comment reads backwards:
I'd expect to just read the condition as "for issuers, we use the account flag otherwise we use the trustline flag".

So something like:
```
        if (header.current().ledgerVersion >= 16)
        {
            bool enableClawback;
            if (getSourceID() == getIssuer(asset))
            {
                enableClawback = isClawbackEnabledOnAccount(sourceAccount);
            }
            else
            {
                enableClawback = trustline.isClawbackEnabled();
            }

            if (enableClawback)
            {
                setClaimableBalanceClawbackEnabled(claimableBalanceEntry);
            }
        }
```

More verbose, but easier to read/reason about and we don't need that comment anymore.

Reply:
Done.